utils/ssh.py

```diff
@@ -456,17 +456,24 @@ def __init__(self, remote_filename, **connect_kwargs):
         self._remote_file_size = None
 
     def __iter__(self):
+        for line in self.raw_lines:
+            yield line.rstrip()
+
+    def raw_lines(self):
         with self as sshtail:
             fstat = sshtail._sftp_client.stat(self._remote_filename)
             if self._remote_file_size is not None:
                 if self._remote_file_size < fstat.st_size:
                     remote_file = self._sftp_client.open(self._remote_filename, 'r')
                     remote_file.seek(self._remote_file_size, 0)
                     while (remote_file.tell() < fstat.st_size):
-                        line = remote_file.readline().rstrip()
+                        line = remote_file.readline()  # Note the  missing rstrip() here!
                         yield line
             self._remote_file_size = fstat.st_size
 
+    def raw_string(self):
+        return ''.join(l for l in self.raw_lines())
```

Comment:
`return ''.join(self)` or if that does not work `return ''.join(self.lines_as_list())`